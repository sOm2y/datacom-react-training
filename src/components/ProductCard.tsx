import React from 'react'
import { connect } from 'react-redux'
import { Card, Button } from 'antd'
import { addCartItem } from '../store/cart/actions'
import { Product } from '../entities/Product';

export const ProductCard = ({ ...props }) => {
  const { product, addProduct } = props
  return (
    <div style={{ background: '#ECECEC', padding: '30px' }}>
      <Card title={product.name} bordered={false}>
        <p>
          {product.price.toLocaleString(undefined, {
            style: 'currency',
            currency: 'NZD',
            minimumFractionDigits: 0,
            maximumFractionDigits: 2
          })}
        </p>
        <Button
          type="primary"
          onClick={() => {
            /**
             * example for adding products to cart
             */
            addProduct(product)
          }}
        >
          Add to cart
        </Button>
      </Card>
    </div>
  )
}

const mapDispatchToProps = {
  addProduct: (product:Product) => addCartItem(product)
}
export default connect(null, mapDispatchToProps)(ProductCard)
