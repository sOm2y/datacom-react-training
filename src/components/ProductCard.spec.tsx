import React from 'react'
import {shallow} from 'enzyme'
import configureStore from 'redux-mock-store'
import ProductCard from './ProductCard'

describe('Component test for <ProductCard />',()=>{
    const mockedStore = {}
    const mockStore = configureStore()
    let store,component:any
    beforeEach(()=>{
        store = mockStore(mockedStore)
        component = shallow(<ProductCard store={store} />)
    })
    it('should display product container component',()=>{
        expect(component.length).toBe(1)
        expect(component).toMatchSnapshot()
    })
})
