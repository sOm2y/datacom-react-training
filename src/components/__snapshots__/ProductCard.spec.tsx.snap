// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test for <ProductCard /> should display product container component 1`] = `ShallowWrapper {}`;
